### PR TITLE
Github Actions -- Daily Production Deploy

### PR DESCRIPTION
**Pull request by oseasmoran73**
## Description

This PR addresses the following story of copying [daily-deploy to github-actions](https://github.com/department-of-veterans-affairs/va.gov-team/issues/24191). Currently being tested against dev and when ready, separate PR will be made to change to prod

## Testing done

Most of testing done in content-build repo. Changes made accordingly

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: script/github-actions/validate-build-status.js [hunk 1]
```diff
@@ -0,0 +1,123 @@
+/* eslint-disable no-console */
+/* eslint-disable camelcase */
+const fetch = require('node-fetch');
```
Comment (oseasmoran73) [inline script/github-actions/validate-build-status.js:3]: This file is the one currently in master for `content-build`. Changes discussed and being made will be transferred over from `content-build` repo

File: .github/workflows/daily-production-deploy.yml [hunk 2]
```diff
@@ -0,0 +1,359 @@
...
+    name: Deploy
+    runs-on: ubuntu-latest
+    needs: create-release
+
+    # TODO: Uncomment when ready for prod
+    env:
+      # ENVIRONMENT: vagovprod
+      # BUCKET: content.www.va.gov
+      ENVIRONMENT: vagovdev
+      BUCKET: content.dev.va.gov
```
Comment (U-DON) [inline .github/workflows/daily-production-deploy.yml:277]:
```suggestion
      # BUCKET: apps.www.va.gov
      ENVIRONMENT: vagovdev
      BUCKET: apps.dev.va.gov
```

File: .github/workflows/daily-production-deploy.yml [hunk 3]
```diff
@@ -0,0 +1,359 @@
...
+      # DRUPAL_ADDRESS: http://internal-dsva-vagov-prod-cms-2000800896.us-gov-west-1.elb.amazonaws.com
+      DRUPAL_ADDRESS: https://cms-vets-website-branch-builds-lo9uhqj18nwixunsjadvjsynuni7kk1u.ci.cms.va.gov
+
+    steps:
+      - name: Checkout vagov-content
+        uses: actions/checkout@v2
+        with:
+          repository: department-of-veterans-affairs/vagov-content
+          path: vagov-content
+
```
Comment (U-DON) [inline .github/workflows/daily-production-deploy.yml:147]:
```suggestion
```

File: .github/workflows/daily-production-deploy.yml [hunk 4]
```diff
@@ -0,0 +1,359 @@
...
+      - name: Checkout vets-website
+        uses: actions/checkout@v2
+        with:
+          path: vets-website
```
Comment (U-DON) [inline .github/workflows/daily-production-deploy.yml:151]:
```suggestion
```

File: .github/workflows/daily-production-deploy.yml [hunk 5]
```diff
@@ -0,0 +1,359 @@
...
+            node_modules
+          key: ${{ steps.get-node-version.outputs.NODE_VERSION }}-${{ hashFiles('yarn.lock') }}
+
+      - name: Install dependencies
+        run: yarn install --frozen-lockfile --prefer-offline
+        env:
+          YARN_CACHE_FOLDER: ~/.cache/yarn
+
+      - name: Build
+        run: yarn build --buildtype=${{ env.BUILDTYPE }} --drupal-address="${{ env.DRUPAL_ADDRESS }}" --no-drupal-proxy
```
Comment (U-DON) [inline .github/workflows/daily-production-deploy.yml:177]:
```suggestion
        run: yarn build --verbose --buildtype=${{ env.BUILDTYPE }}
```

File: .github/workflows/daily-production-deploy.yml [hunk 6]
```diff
@@ -0,0 +1,359 @@
...
+      run:
+        working-directory: vets-website
+
+    # TODO: Uncomment when ready for prod
+    env:
+      NODE_EXTRA_CA_CERTS: /etc/ssl/certs/VA-Internal-S2-RCA1-v1.cer.pem
+      BUILDTYPE: vagovprod
+      CHROMEDRIVER_FILEPATH: /usr/local/share/chrome_driver/chromedriver
+      # DRUPAL_ADDRESS: http://internal-dsva-vagov-prod-cms-2000800896.us-gov-west-1.elb.amazonaws.com
+      DRUPAL_ADDRESS: https://cms-vets-website-branch-builds-lo9uhqj18nwixunsjadvjsynuni7kk1u.ci.cms.va.gov
```
Comment (U-DON) [inline .github/workflows/daily-production-deploy.yml:139]:
```suggestion
```

**Review by U-DON — COMMENTED** [review]
Still has some things left over that are related to `content-build`.

File: .github/workflows/daily-production-deploy.yml [hunk 7]
```diff
@@ -0,0 +1,359 @@
...
+      - name: Install dependencies
+        run: yarn install --frozen-lockfile --prefer-offline
+        env:
+          YARN_CACHE_FOLDER: ~/.cache/yarn
+
+      - name: Build
+        run: yarn build --buildtype=${{ env.BUILDTYPE }} --drupal-address="${{ env.DRUPAL_ADDRESS }}" --no-drupal-proxy
+        timeout-minutes: 30
+        env:
+          NODE_ENV: production
```
Comment (U-DON) [inline .github/workflows/daily-production-deploy.yml:180]:
```suggestion
```

File: .github/workflows/daily-production-deploy.yml [hunk 8]
```diff
@@ -0,0 +1,359 @@
...
+          attachments: '[{"mrkdwn_in": ["text"], "color": "good", "text": "Stand by, production deploy for vets-website coming up in ${{ needs.set-env.outputs.RELEASE_WAIT }} minutes. View what coming here: <https://github.com/${{github.repository}}/compare/vets-website/${{needs.set-env.outputs.LATEST_TAG_VERSION}}...${{github.sha}}>"}]'
+          channel_id: ${{ env.CHANNEL_ID }}
+  
+  build:
+    name: Build
+    runs-on: self-hosted
+    needs: notify-start
+    defaults:
+      run:
+        working-directory: vets-website
```
Comment (U-DON) [inline .github/workflows/daily-production-deploy.yml:131]:
```suggestion
```

**Review by U-DON — APPROVED** [review]

**Pull request merged**